csharp/src/Apache.Arrow.Adbc/Extensions/MarshalExtensions.netstandard.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if NETSTANDARD
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Apache.Arrow.Adbc.Extensions
+{
+    public static class MarshalExtensions
+    {
+        public static unsafe string PtrToStringUTF8(IntPtr intPtr)
+        {
+            if (intPtr == null) 
+            {
+                return null;
+            }
+
+            unsafe
+            {
+                byte* source = (byte*)intPtr;
+                int nbBytes =  source[0];

```

Comment:
Is this right? 

1. These strings have the length at the beginning? Is there some guarantee about that?
2. Since `source` is a `byte*`, `source[0]` returns a `byte`, which means the `nbBytes` can only be between 0-255. These strings can't be longer than that?

I would have assumed we would need to `strlen` here - like the real Marshal.PtrToStringUTF8 does - https://github.com/dotnet/runtime/blob/7febca6802545d8c96e386187d536bc100d40337/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs#L76

Reply:
added a different approach